### PR TITLE
[re_renderer] Introduce `DrawPhase`es for simplistic multi pass rendering

### PR DESCRIPTION
**Pull request by Wumpf**
Or rather as a first and super easy step towards it. Also takes the first step towards supporting transparency, see #702 Replaces previous sorting key!

Very simple idea: Every `Renderer` is invoked with every `DrawPhase` it registered for. `DrawPhase` have a strict order and implicitly describe the `wgpu::RenderPass` that is passed for drawing.
Nice perk is that our previously very special composition renderer is now just another regular renderer - the only one interested in the `Composition` phase!

For now and forseeable future, these passes are managed by the ViewBuilder (unchanged!). At some point we will probably make `DrawPhase` a more complex object, but this is not today!

### Checklist
* [x] I have read and agree to [Contributor Guide](../CONTRIBUTING.md) and the [Code of Conduct](../CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)
* [ ] I've added a line to `CHANGELOG.md` (if this is a big enough change to warrant it)

### CODE REVIEW

File: crates/re_renderer/src/renderer/mod.rs [hunk 1]
```diff
@@ -52,45 +52,36 @@ pub trait Renderer {
...
 
-    /// Backgrounds should always be rendered last.
-    Background = 70000,
+    /// Background, rendering where depth wasn't written.
+    Background,
 
-    /// Postprocessing effects that are applied before the final tonemapping step.
-    Postprocess = 90000,
+    /// Drawn when compositing with the main target.
+    Composition,
```
Comment (emilk) [inline crates/re_renderer/src/renderer/mod.rs:86]:
```suggestion
    Compositing,
```

composing != compositing :)

File: crates/re_renderer/src/view_builder.rs [hunk 2]
```diff
@@ -423,25 +418,39 @@ impl ViewBuilder {
...
+    fn draw_phase<'a>(
+        &'a self,
+        ctx: &'a RenderContext,
+        phase: DrawPhase,
+        pass: &mut wgpu::RenderPass<'a>,
+    ) -> Result<(), anyhow::Error> {
+        for queued_draw in &self.queued_draws {
+            if queued_draw.participated_phases.contains(&phase) {
+                (queued_draw.draw_func)(ctx, phase, pass, queued_draw.draw_data.as_ref())
+                    .context("drawing a view")?;
```
Comment (emilk) [inline crates/re_renderer/src/view_builder.rs:430]: this context could mention the phase, and maybe even the name of the `queued_draw`

**Review by emilk — APPROVED** [review]
Very nice!

Some day soon we need to consider how we express general render-graphcs

File: crates/re_renderer/src/view_builder.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_renderer/src/view_builder.rs:430]: adding phase :). name is a bit too cumbersome now

**Comment by Wumpf** [comment]
oops, added a wip file. reverting!

**Pull request merged**